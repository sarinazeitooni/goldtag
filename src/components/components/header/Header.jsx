import React from 'react';
import Navbar from './navbar/Navbar';
import texts from './texts/texts';
import './style/header.scss';
import './style/location.scss';
import './style/search.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faShoppingBag, faShoppingBasket ,faSearch , faUserFriends ,faMapMarker ,faBars}
from "@fortawesome/free-solid-svg-icons";
import { BrowserView, MobileView} from "react-device-detect";
function Header() {
    return (
      <div className="header">
          <BrowserView>
          <div className="header-top-container">
            <div className="search-container">
              <button className="basket">
                <FontAwesomeIcon className="icon" icon={faShoppingBag}/>
                {texts.basket}</button>
              <div className="search-box">
              <FontAwesomeIcon style={{position: "absolute",top: "9px", right: "5px"}} className="icon" icon={faSearch}/>
              <input className="search-input" type="text"
              placeholder={texts.inputPlaceHolder}></input>
                <button className="search-btn">{texts.search}</button>
              </div>
            </div>
            <div className="location-container">
              <button className="location-btn">
                <span>{texts.location}</span>
                <FontAwesomeIcon style={{ marginRight : "14px" , color : 'rgba(190,146,67,.95)'}} className="icon" icon={faMapMarker}/>
                </button>
              <img className="goldtagIcon" alt="golTagIcon" src="https://cdn.goldtag.net/frontend/assets/v1/desktop/build/images/logo.svg"></img>
            </div>
          </div>
          <hr></hr>
        </BrowserView>
          <MobileView>
            <div className="header-top-container">
            <FontAwesomeIcon className="header-item icon ham-menu" icon={faBars}/>
            <FontAwesomeIcon className="header-item icon" icon={faSearch}/>
            <img alt="goldtag" src="https://cdn.goldtag.net/frontend/assets/v1/mobile/build/images/mobile-logo.svg"></img>
            <FontAwesomeIcon className="header-item icon" icon={faUserFriends}/>
            <div className=" header-item basket-container">
              <div className="item-numbers">0</div>
            <FontAwesomeIcon className="header-item icon" icon={faShoppingBasket}/>
            </div>
            </div>       
          </MobileView>
          <BrowserView>
          <Navbar/>
          </BrowserView>
      </div>
    );
  }
  export default Header;
  