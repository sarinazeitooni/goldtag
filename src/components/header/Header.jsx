import React from 'react';
import Navbar from './navbar/Navbar';
import headerTexts from './texts/headerTexts';
import './style/header.scss';
import './style/location.scss';
import './style/search.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faShoppingBag ,faSearch ,faMapMarker} from "@fortawesome/free-solid-svg-icons";
import {BrowserView} from "react-device-detect";
import Divider from "../divider/Divider";
import ResponsiveHeader from "./ResponsiveHeader";
function Header() {
    return (
        <React.Fragment>
          <div className="header">
          <BrowserView>
          <div className="header-top-container">
            <div className="search-container">
              <button className="basket">
                <FontAwesomeIcon className="icon" icon={faShoppingBag}/>{headerTexts.basket}</button>
              <div className="search-box">
              <FontAwesomeIcon className="icon search-icon" icon={faSearch}/>
              <input className="search-input" type="text"
              placeholder={headerTexts.inputPlaceHolder}/>
                <button className="search-btn">{headerTexts.search}</button>
              </div>
            </div>
            <div className="location-container">
              <button className="location-btn">
                <span>{headerTexts.location}</span>
                <FontAwesomeIcon className="icon map-icon" icon={faMapMarker}/>
                </button>
              <img className="goldtagIcon" alt="golTagIcon" src="https://cdn.goldtag.net/frontend/assets/v1/desktop/build/images/logo.svg"/>
            </div>
          </div>
          <Divider className=''/>
        </BrowserView>
      </div>
          <ResponsiveHeader/>
          <Navbar/>
          </React.Fragment>
    );
  }
  export default Header;
  