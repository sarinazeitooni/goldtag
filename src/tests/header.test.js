import React from 'react';
import {shallow} from 'enzyme';
import Header from "../components/header/Header";
import ReponsiveHeader from "../components/header/ResponsiveHeader";
describe("header component" , ()=>{
    it("header should match snap shot" , ()=>{
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    });
    it("responsive header should match snap shot" , ()=>{
        const wrapper = shallow(<ReponsiveHeader/>);
        expect(wrapper).toMatchSnapshot();
    });
    it("header should render src correctly" , ()=>{
        const wrapper = shallow(<Header/>);
        const image = wrapper.find('img.goldtagIcon');
        expect(image.prop("src")).toEqual("https://cdn.goldtag.net/frontend/assets/v1/desktop/build/images/logo.svg");
    });
    it("responsive header should render src correctly" , ()=>{
        const wrapper = shallow(<ReponsiveHeader/>);
        const image = wrapper.find('img.goldtag');
        expect(image.prop("src")).toEqual("https://cdn.goldtag.net/frontend/assets/v1/mobile/build/images/mobile-logo.svg");
    });
})