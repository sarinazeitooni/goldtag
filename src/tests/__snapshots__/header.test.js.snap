// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component header should match snap shot 1`] = `ShallowWrapper {}`;

exports[`header component responsive header should match snap shot 1`] = `ShallowWrapper {}`;
